internal/boxcli/usererr/exiterr.go

```diff
@@ -1,12 +1,12 @@
 package usererr
 
 import (
-	"os/exec"
-
 	"errors"
+	"os/exec"
 )
 
-type UserExecError struct {
+// ExitError is an ExitError for a command run on behalf of a user
+type ExitError struct {
```

Comment:
I'm curious why we have a new type for this. Why not just use `exec.ExitError` directly?

Reply:
@gcurtis the idea is that a user exit error and a generic exit error are treated differently in terms of what we show the user. For errors that were a result of an error caused by user code we want to always show as much info as possible. For non-user exit errors we only show all info if DEVBOX_DEBUG is true.

Reply:
This type is used for errors that are caused by the user's program, as opposed to a devbox CLI error. For example, when doing `devbox run -- <program>` the user's program may exit with an error. So, if we can detect that circumstance, we wrap the `exec.Command`'s error in this special type and then render the error message accordingly.